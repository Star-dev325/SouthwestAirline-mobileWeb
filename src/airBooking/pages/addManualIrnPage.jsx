// @flow
import React from 'react';
import _ from 'lodash';

import { updateSelectedIrn } from 'src/airBooking/actions/airBookingActions';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import withBodyClass from 'src/shared/enhancers/withBodyClass';
import { connect } from 'react-redux';
import AddManualIrnForm from 'src/airBooking/components/addManualIrnForm';
import { AIR_BOOKING_ADD_MANUAL_IRN } from 'src/shared/constants/formIds';
import { raiseSatelliteEvent } from 'src/shared/analytics/helpers/analyticsEventHelper';

import type { Push } from 'src/shared/flow-typed/shared.types';
import type { ManualIrnFormDataType, SelectedIrnType } from 'src/airBooking/flow-typed/airBooking.types';

type AddManualIrnPageProps = {
  goBack: () => void,
  push: Push,
  selectedIrn?: SelectedIrnType,
  updateSelectedIrnFn: (string, ?boolean) => void
};

export class AddManualIrnPage extends React.Component<AddManualIrnPageProps> {
  componentDidMount = () => {
    raiseSatelliteEvent('add irn');
  };

  _onSubmit = (formData: ManualIrnFormDataType) => {
    const { push, updateSelectedIrnFn } = this.props;

    updateSelectedIrnFn(formData.manualIrn, true);
    push(getNormalizedRoute({ routeName: 'purchaseWithoutClearForm' }));
  };

  render() {
    const { goBack, selectedIrn } = this.props;

    return (
      <div className="add-manual-internal-reference-number-page">
        <AddManualIrnForm
          onSubmit={this._onSubmit}
          formId={AIR_BOOKING_ADD_MANUAL_IRN}
          onCancel={goBack}
          selectedIrn={selectedIrn}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  selectedIrn: _.get(state, 'app.airBooking.selectedIrn')
});

const mapDispatchToProps = {
  updateSelectedIrnFn: updateSelectedIrn
};

const enhancers = _.flowRight(
  withBodyClass('hide-header'),
  withConnectedReactRouter,
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(AddManualIrnPage);
