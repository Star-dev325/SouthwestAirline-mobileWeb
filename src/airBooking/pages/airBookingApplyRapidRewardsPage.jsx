// @flow
import i18n from '@swa-ui/locale';
import _ from 'lodash';
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import * as AirBookingActions from 'src/airBooking/actions/airBookingActions';
import ApplyRapidRewardsForm from 'src/airBooking/components/applyRapidRewardsForm';
import { transformToCalculateSplitPayCalcFundsRequest } from 'src/airBooking/transformers/applyRapidRewardsTransformer';
import * as ApplyTravelFundsActions from 'src/shared/actions/applyTravelFundsActions';
import * as DialogActions from 'src/shared/actions/dialogActions';
import * as FormDataActions from 'src/shared/actions/formDataActions';
import { raiseSatelliteEvent } from 'src/shared/analytics/helpers/analyticsEventHelper';
import Button from 'src/shared/components/button';
import FundResultsList from 'src/shared/components/fundResultsList';
import PageHeaderWithButtons from 'src/shared/components/pageHeaderWithButtons';
import PriceTotal from 'src/shared/components/priceTotal';
import Segment from 'src/shared/components/segment';
import { DESTRUCTIVE } from 'src/shared/constants/buttonPopupStyleTypes';
import { AIR_BOOKING_APPLY_RAPID_REWARDS_FORM } from 'src/shared/constants/formIds';
import withBodyClass from 'src/shared/enhancers/withBodyClass';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import withHideGlobalHeader from 'src/shared/enhancers/withHideGlobalHeader';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';
import TravelFundsConstants from 'src/travelFunds/constants/travelFundsConstants';
import { transformToRemoveFundsRequest } from 'src/travelFunds/transformers/travelFundsTransformer';
import DynamicPlacement from 'src/wcm/components/dynamicPlacement';

import type {
  ApplySplitPayPageCalcFundsResponse,
  SplitPayCalcFundsRequestType,
  SplitPayMessageType,
  SplitPayOptionsListRequestType,
  SplitPayPageResponseType,
  SplitPayPageSuccessType,
  TotalPointsAppliedType,
  UserNameInfo
} from 'src/airBooking/flow-typed/applyRapidRewards.types';
import type { RemoveFundRequestType } from 'src/airBooking/flow-typed/calcFunds.types';
import type { PassengerInfos } from 'src/shared/flow-typed/shared.types';
import type { DynamicPlacementResponse } from 'src/wcm/flow-typed/wcm.types';

const { SPLIT_PAYMENT } = TravelFundsConstants;

type Props = {
  accountRedeemablePoints: number,
  applySplitPayPageCalcFundsResponse: ?ApplySplitPayPageCalcFundsResponse,
  calculateFundsFn: (SplitPayCalcFundsRequestType, ?string, ?boolean) => void,
  clearFormDataByIdFn: (string) => {},
  goBack: () => void,
  hideDialogFn: () => Promise<*>,
  isLoggedIn: boolean,
  isWebView: boolean,
  itineraryPricingToken: string,
  loadSplitPayPagePlacementsFn: (*) => void,
  onSubmitApplyRapidRewardPoints: (formData: *) => void,
  passengerInfos: PassengerInfos,
  radioOptionSelected?: string,
  removeFundFn: (RemoveFundRequestType, ?string, ?boolean) => void,
  resetCalculateFlowDataFn: () => void,
  resetSplitPayTermsAndConditionsFn: () => void,
  showDialogFn: (*) => Promise<*>,
  splitPayMessage?: SplitPayMessageType,
  splitPayOptionsSecureRequestObj: SplitPayOptionsListRequestType,
  splitPayPage: SplitPayPageSuccessType,
  splitPayPagePlacements: { paymentBanner: DynamicPlacementResponse },
  splitPayPageResponse: SplitPayPageResponseType,
  termsAndConditions: ?string,
  totalPointsApplied?: ?TotalPointsAppliedType,
  userNameInfo: UserNameInfo
};

export const AirBookingApplyRapidRewardsPage = ({
  accountRedeemablePoints,
  applySplitPayPageCalcFundsResponse,
  calculateFundsFn,
  clearFormDataByIdFn,
  goBack,
  hideDialogFn,
  isLoggedIn,
  isWebView,
  itineraryPricingToken,
  loadSplitPayPagePlacementsFn,
  passengerInfos,
  radioOptionSelected,
  removeFundFn,
  resetCalculateFlowDataFn,
  resetSplitPayTermsAndConditionsFn,
  showDialogFn,
  splitPayMessage,
  splitPayPage,
  splitPayPagePlacements,
  splitPayPageResponse,
  termsAndConditions,
  userNameInfo
}: Props) => {
  const { _links, splitPayOptions } = splitPayPage || {};
  const { balanceRemaining } = splitPayPageResponse || {};
  const { fundsAppliedToken, selectedSplitPay, totalFunds, totalPointsApplied, travelFunds } = applySplitPayPageCalcFundsResponse || {};
  const { paymentBanner } = splitPayPagePlacements || {};
  const applyRapidRewardsUrl = getNormalizedRoute({ routeName: 'applyRapidRewards' });
  const DEFAULT_REDIRECT_PATH = '/air/booking';
  const splitPayOptionsExists = splitPayOptions && splitPayOptions.length > 0;
  const totals = applySplitPayPageCalcFundsResponse?.totals ?? splitPayPageResponse?.totals;

  const priceTotalProps = {
    shouldHidePriceBreakdown: true,
    totalAppliedTravelFunds: applySplitPayPageCalcFundsResponse?.totalFunds ?? totalFunds,
    totalPointsApplied,
    totals: totals,
    travelFundsBalanceRemaining: applySplitPayPageCalcFundsResponse?.balanceRemaining ?? balanceRemaining
  };

  useEffect(() => {
    loadSplitPayPagePlacementsFn();
  }, []);

  const onSubmitApplyRapidRewardPoints = (formData) => {
    const splitPayCalcFundsRequest = transformToCalculateSplitPayCalcFundsRequest(
      formData,
      fundsAppliedToken,
      itineraryPricingToken,
      passengerInfos,
      userNameInfo
    );

    calculateFundsFn(splitPayCalcFundsRequest, DEFAULT_REDIRECT_PATH, isLoggedIn);
    raiseSatelliteEvent('squid', { page_description: 'button:cash plus points apply points' });
  };

  const getSplitPayFundId = () => {
    const splitPayFund = travelFunds && travelFunds.filter((fund) => fund.travelFundType === SPLIT_PAYMENT);

    return (splitPayFund && splitPayFund[0]?._links?.removeTravelFund?.body?.removalTravelFundId) ?? '0';
  };

  const removeSplitPayFund = async (fundId: string) => {
    const removeSPFundRequest = transformToRemoveFundsRequest(
      fundId,
      passengerInfos,
      fundsAppliedToken,
      itineraryPricingToken,
      true
    );
    const splitPayFundId = getSplitPayFundId();

    if (fundId === splitPayFundId) {
      clearFormDataByIdFn(AIR_BOOKING_APPLY_RAPID_REWARDS_FORM);
    }

    travelFunds &&
      travelFunds.length > 1 &&
      (await removeFundFn(removeSPFundRequest, applyRapidRewardsUrl, isLoggedIn));

    if (travelFunds && travelFunds.length === 1) {
      resetCalculateFlowDataFn();
      resetSplitPayTermsAndConditionsFn();
    }
  };

  const splitPayCancel = async (fundRemovalId) => {
    await removeSplitPayFund(fundRemovalId);
    goBack();
  };

  const onClickCancelButton = () => {
    if (totalPointsApplied) {
      const fundRemovalId = getSplitPayFundId();

      showDialogFn({
        buttons: [
          {
            label: i18n('SHARED__BUTTON_TEXT__CANCEL'),
            onClick: hideDialogFn
          },
          {
            label: i18n('SHARED__BUTTON_TEXT__OK'),
            onClick: () => {
              hideDialogFn();
              splitPayCancel(fundRemovalId);
            },
            style: DESTRUCTIVE
          }
        ],
        message: i18n('SPLIT_PAY_PAGE__CANCEL_DIALOG_MESSAGE'),
        name: 'SPLIT_PAY_PAGE__CANCEL_DIALOG',
        title: i18n('SPLIT_PAY_PAGE__CANCEL_DIALOG_TITLE')
      });
    } else {
      goBack();
    }
  };

  const rightButton = [
    {
      name: i18n('SHARED__BUTTON_TEXT__CANCEL'),
      onClick: onClickCancelButton
    }
  ];

  return (
    <div className="apply-rapid-rewards-page">
      {<PageHeaderWithButtons rightButtons={rightButton} title={i18n('SPLIT_PAY_PAGE__TITLE')} />}
      {splitPayOptionsExists && (
        <>
          {paymentBanner && (
            <DynamicPlacement
              {...paymentBanner}
              additionalTemplateData={accountRedeemablePoints}
              data-qa="paymentBanner"
              isWebView={isWebView}
            />
          )}
          <ApplyRapidRewardsForm
            formId={AIR_BOOKING_APPLY_RAPID_REWARDS_FORM}
            links={_links}
            onSubmit={onSubmitApplyRapidRewardPoints}
            radioOptionSelected={radioOptionSelected}
            selectedSplitPay={selectedSplitPay}
            splitPayRadioOptions={splitPayOptions}
            termsAndConditions={termsAndConditions}
            totalPointsApplied={totalPointsApplied}
          />
        </>
      )}
      {!splitPayOptionsExists && splitPayMessage && (
        <div className="split-pay-message" dangerouslySetInnerHTML={{ __html: splitPayMessage.body }} />
      )}
      {travelFunds && (
        <div className="apply-travel-funds--results">
          <FundResultsList
            listTitle={i18n('SHARED__FUND_RESULT__ELIGIBLE_FUNDS')}
            removeFundFn={removeSplitPayFund}
            retrievedFunds={travelFunds}
          />
        </div>
      )}
      {(totalPointsApplied || travelFunds) && totals && (
        <Segment className="purchase-content--summary-footer">
          <PriceTotal {...priceTotalProps} />
          <div className="purchase-content--summary-footer-nav">
            <Button className="apply-continue-button purchase" fluid onClick={goBack} size="larger" type="button">
              {i18n('SHARED__BUTTON_TEXT__CONTINUE')}
            </Button>
          </div>
        </Segment>
      )}
    </div>
  );
};

const mapStateToProps = (state) => ({
  accountRedeemablePoints: state?.app?.account?.accountInfo?.rapidRewardsDetails?.redeemablePoints,
  applySplitPayPageCalcFundsResponse: state?.app?.airBooking?.applyTravelFundsPage?.response,
  itineraryPricingToken:
    state?.app?.airBooking?.splitPay?.splitPayPage?._links?.calculateFunds?.body?.itineraryPricingToken ||
    state?.app?.airBooking?.flightPricingPage?.response?.flightPricingPage?._links?.calculateFunds?.body
      ?.itineraryPricingToken,
  passengerInfos: state?.app?.airBooking?.passengerInfos,
  radioOptionSelected: state?.app?.formData?.AIR_BOOKING_APPLY_RAPID_REWARDS_FORM?.data?.selectedRadioOption,
  splitPayMessage: state?.app?.airBooking?.splitPay?.splitPayMessage,
  splitPayPage: state?.app?.airBooking?.splitPay?.splitPayPage,
  splitPayPagePlacements: state?.app?.airBooking?.splitPayPagePlacements,
  splitPayPageResponse: state?.app?.airBooking?.splitPay,
  termsAndConditions: state?.app?.airBooking?.splitPayTermsAndConditions,
  userNameInfo: state?.app?.account?.accountInfo?.customerInfo?.name
});

const mapDispatchToProps = {
  calculateFundsFn: ApplyTravelFundsActions.calculateFunds,
  clearFormDataByIdFn: FormDataActions.clearFormDataById,
  hideDialogFn: DialogActions.hideDialog,
  loadSplitPayPagePlacementsFn: AirBookingActions.loadSplitPayPagePlacements,
  removeFundFn: ApplyTravelFundsActions.removeFund,
  resetCalculateFlowDataFn: ApplyTravelFundsActions.resetCalculateFlowData,
  resetSplitPayTermsAndConditionsFn: AirBookingActions.resetSplitPayTermsAndConditions,
  showDialogFn: DialogActions.showDialog
};

const enhancers = _.flowRight(
  withBodyClass('air-booking-apply-rapid-rewards-page'),
  withConnectedReactRouter,
  withHideGlobalHeader,
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(AirBookingApplyRapidRewardsPage);
