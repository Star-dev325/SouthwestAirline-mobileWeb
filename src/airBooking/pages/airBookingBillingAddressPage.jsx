// @flow
import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import * as AirBookingActions from 'src/airBooking/actions/airBookingActions';
import BillingAddressForm from 'src/shared/components/billingAddressForm';
import withHideGlobalHeader from 'src/shared/enhancers/withHideGlobalHeader';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import { transformContactInfoToBillingAddressFormData } from 'src/shared/transformers/billingAddressTransformer';
import { AIRBOOKING_BILLING_ADDRESS_FORM } from 'src/shared/constants/formIds';

import type { FormData } from 'src/shared/form/flow-typed/form.types';
import type { Push, AccountContactInfoType, BillingAddressFormType } from 'src/shared/flow-typed/shared.types';

type Props = {
  push: Push,
  goBack: () => void,
  contactInfo?: AccountContactInfoType,
  travelFundsAddress?: BillingAddressFormType,
  saveTravelFundsBillingAddressFn: (BillingAddressFormType) => void
};

export class AirBookingBillingAddressPage extends React.Component<Props> {
  _getInitialFormData = () => {
    const { contactInfo, travelFundsAddress } = this.props;

    if (contactInfo && _.isEmpty(travelFundsAddress)) {
      return transformContactInfoToBillingAddressFormData(contactInfo);
    } else if (travelFundsAddress) {
      return travelFundsAddress;
    }
  };

  _onSubmit = (formData: FormData) => {
    const { goBack, saveTravelFundsBillingAddressFn } = this.props;

    saveTravelFundsBillingAddressFn(formData);
    goBack();
  };

  render() {
    const { goBack } = this.props;
    const initialFormData = this._getInitialFormData();

    return (
      <div>
        <BillingAddressForm
          initialFormData={initialFormData}
          formId={AIRBOOKING_BILLING_ADDRESS_FORM}
          onSubmit={this._onSubmit}
          goBack={goBack}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  contactInfo: _.get(state, 'app.account.accountInfo.contactInfo'),
  travelFundsAddress: _.get(state, 'app.airBooking.purchaseSummaryPage.travelFundsAddress')
});

const mapDispatchToProps = {
  saveTravelFundsBillingAddressFn: AirBookingActions.saveTravelFundsBillingAddress
};

const enhancers = _.flowRight(
  withConnectedReactRouter,
  withHideGlobalHeader,
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(AirBookingBillingAddressPage);
