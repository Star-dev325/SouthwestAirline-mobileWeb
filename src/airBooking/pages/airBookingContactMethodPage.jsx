// @flow
import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import withHideGlobalHeader from 'src/shared/enhancers/withHideGlobalHeader';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import * as AirBookingActions from 'src/airBooking/actions/airBookingActions';
import { isAlreadyHasContactMethod } from 'src/airBooking/selectors/contactMethodPageSelectors';
import ContactMethodPage from 'src/shared/pages/contactMethodPage';
import { AIR_BOOKING_CONTACT_METHOD_FORM } from 'src/shared/constants/formIds';
import withBodyClass from 'src/shared/enhancers/withBodyClass';

import type { ContactMethodInfo } from 'src/shared/flow-typed/shared.types';

type Props = {
  updateContactMethodFn: (*) => void,
  goBack: () => void,
  contactMethodInfo: ?ContactMethodInfo,
  isInternationalBooking: boolean,
  alreadyHasContactMethod: boolean,
  isLoggedIn: boolean
};

export class AirBookingContactMethodPage extends React.Component<Props> {
  render() {
    const {
      updateContactMethodFn,
      goBack,
      contactMethodInfo,
      isInternationalBooking,
      alreadyHasContactMethod,
      isLoggedIn
    } = this.props;

    return (
      <ContactMethodPage
        formId={AIR_BOOKING_CONTACT_METHOD_FORM}
        updateContactMethodFn={updateContactMethodFn}
        goBack={goBack}
        contactMethodInfo={contactMethodInfo}
        isInternationalBooking={isInternationalBooking}
        isAlreadyHasContactMethod={alreadyHasContactMethod}
        isLoggedIn={isLoggedIn}
        isAirBooking
      />
    );
  }
}

const mapDispatchToProps = {
  updateContactMethodFn: AirBookingActions.updateContactMethod
};

const mapStateToProps = (state) => ({
  alreadyHasContactMethod: isAlreadyHasContactMethod(state),
  contactMethodInfo: _.get(state, 'app.airBooking.contactMethodInfo'),
  isInternationalBooking: _.get(state, 'app.airBooking.isInternationalBooking'),
  isLoggedIn: _.get(state, 'app.account.isLoggedIn')
});

const enhancers = _.flowRight(
  withConnectedReactRouter,
  withHideGlobalHeader,
  connect(mapStateToProps, mapDispatchToProps),
  withBodyClass('air-booking-contact-method-page')
);

export default enhancers(AirBookingContactMethodPage);
