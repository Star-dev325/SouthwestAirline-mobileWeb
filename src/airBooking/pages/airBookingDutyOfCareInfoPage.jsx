// @flow
import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import withHideGlobalHeader from 'src/shared/enhancers/withHideGlobalHeader';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import * as AirBookingActions from 'src/airBooking/actions/airBookingActions';
import ContactInfoTravelManagerPage from 'src/shared/pages/contactInfoTravelManagerPage';
import { AIR_BOOKING_INFO_CONTACT_TRAVEL_MANAGER_FORM } from 'src/shared/constants/formIds';
import withBodyClass from 'src/shared/enhancers/withBodyClass';

import type { DutyOfCare } from 'src/shared/flow-typed/shared.types';
import { updateFormFieldDataValue } from 'src/shared/actions/formDataActions';

type Props = {
  updateContactInfoTravelManagerFn: (*) => void,
  goBack: () => void,
  dutyOfCareContact: DutyOfCare,
  isInternationalBooking: boolean,
  isLoggedIn: boolean,
  updateFormFieldDataValueFn: (*) => void,
  contactTravelManagerInfo: DutyOfCare
};

export class AirBookingDutyOfCareInfoPage extends React.Component<Props> {
  render() {
    const {
      updateContactInfoTravelManagerFn,
      goBack,
      dutyOfCareContact,
      isInternationalBooking,
      isLoggedIn,
      updateFormFieldDataValueFn,
      contactTravelManagerInfo
    } = this.props;

    return (
      <ContactInfoTravelManagerPage
        testId="ContactInfoTravelManagerPage"
        formId={AIR_BOOKING_INFO_CONTACT_TRAVEL_MANAGER_FORM}
        updateContactInfoTravelManagerFn={updateContactInfoTravelManagerFn}
        goBack={goBack}
        dutyOfCareContact={contactTravelManagerInfo || dutyOfCareContact}
        isInternationalBooking={isInternationalBooking}
        isLoggedIn={isLoggedIn}
        updateFormFieldDataValueFn={updateFormFieldDataValueFn}
        disclaimerText={dutyOfCareContact?.disclaimerText || ''}
        isAirBooking
      />
    );
  }
}

const mapDispatchToProps = {
  updateContactInfoTravelManagerFn: AirBookingActions.updateContactTravelInfoMethod,
  updateFormFieldDataValueFn: updateFormFieldDataValue
};

const mapStateToProps = (state) => ({
  dutyOfCareContact: _.get(state, 'app.airBooking.accountInfo.dutyOfCareContact'),
  isInternationalBooking: _.get(state, 'app.airBooking.isInternationalBooking'),
  isLoggedIn: _.get(state, 'app.account.isLoggedIn'),
  contactTravelManagerInfo: _.get(state, 'app.airBooking.contactTravelInfo')
});

const enhancers = _.flowRight(
  withConnectedReactRouter,
  withHideGlobalHeader,
  connect(mapStateToProps, mapDispatchToProps),
  withBodyClass('air-booking-contact-info-travel-manager-page')
);

export default enhancers(AirBookingDutyOfCareInfoPage);
