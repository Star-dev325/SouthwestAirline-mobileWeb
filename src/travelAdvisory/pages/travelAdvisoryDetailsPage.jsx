// @flow
import React, { Component } from 'react';
import _ from 'lodash';
import { connect } from 'react-redux';

import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import TravelAdvisoryDetails from 'src/travelAdvisory/components/travelAdvisoryDetails';
import i18n from '@swa-ui/locale';
import NavItem from 'src/shared/components/navItem';
import * as TravelAdvisoryActions from 'src/travelAdvisory/actions/travelAdvisoryActions';

import type { TravelAdvisoryType } from 'src/travelAdvisory/flow-typed/travelAdvisory.types';
import type { Push } from 'src/shared/flow-typed/shared.types';

type Props = {
  params: {
    number: string
  },
  getTravelAdvisoriesFn: () => Promise<*>,
  travelAdvisories?: Array<TravelAdvisoryType>,
  push: Push
};

export class TravelAdvisoryDetailsPage extends Component<Props> {
  static defaultProps = {
    travelAdvisories: [
      {
        id: '',
        advisoryTitle: '',
        advisoryInfo: '',
        stationInfo: []
      }
    ]
  };

  componentDidMount() {
    const { getTravelAdvisoriesFn } = this.props;

    getTravelAdvisoriesFn();
  }

  _showFlightStatus = () => {
    this.props.push('/flight-status');
  };

  _showMyTrips = () => {
    this.props.push('/my-account/upcoming-trips');
  };

  render() {
    const {
      params: { number },
      travelAdvisories
    } = this.props;
    const advisoryDetail = travelAdvisories ? travelAdvisories[parseInt(number)] : false;

    return (
      <div>
        {advisoryDetail && <TravelAdvisoryDetails travelAdvisory={advisoryDetail} />}
        <div>
          <NavItem className="travel-advisory-detail--route-item" onClick={this._showFlightStatus}>
            <span className="travel-advisory-detail--flight-status">
              {i18n('TRAVEL_ADVISORY__FLIGHT_STATUS_TITLE')}
            </span>
          </NavItem>
          <NavItem className="travel-advisory-detail--route-item" onClick={this._showMyTrips}>
            <span className="travel-advisory-detail--my-trips">{i18n('TRAVEL_ADVISORY__MY_TRIPS_TITLE')}</span>
          </NavItem>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  travelAdvisories: state.app.travelAdvisory.messageTravelAdvisory
});

const mapDispatchToProps = {
  getTravelAdvisoriesFn: TravelAdvisoryActions.getTravelAdvisories
};

const enhancers = _.flowRight(withConnectedReactRouter, connect(mapStateToProps, mapDispatchToProps));

export default enhancers(TravelAdvisoryDetailsPage);
