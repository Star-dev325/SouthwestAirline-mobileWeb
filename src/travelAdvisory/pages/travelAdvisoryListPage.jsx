// @flow
import React, { Component } from 'react';
import _ from 'lodash';
import { connect } from 'react-redux';

import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import TravelAdvisoryItem from 'src/travelAdvisory/components/travelAdvisoryItem';
import { buildPathWithParamAndQuery } from 'src/shared/helpers/pathUtils';
import * as TravelAdvisoryActions from 'src/travelAdvisory/actions/travelAdvisoryActions';

import type { TravelAdvisoryType } from 'src/travelAdvisory/flow-typed/travelAdvisory.types';
import type { Push } from 'src/shared/flow-typed/shared.types';

type Props = {
  getTravelAdvisoriesFn: () => Promise<*>,
  travelAdvisories: Array<TravelAdvisoryType>,
  push: Push
};

export class TravelAdvisoryListPage extends Component<Props> {
  static defaultProps = {
    travelAdvisories: [
      {
        id: '',
        advisoryTitle: '',
        advisoryInfo: '',
        stationInfo: []
      }
    ]
  };

  componentDidMount() {
    const { getTravelAdvisoriesFn } = this.props;

    getTravelAdvisoriesFn();
  }

  _showTravelAdvisoryDetails = (index: number) => {
    const nextPagePath = buildPathWithParamAndQuery('/travel-advisories/:number', { number: index });

    this.props.push(nextPagePath);
  };

  render() {
    const { travelAdvisories } = this.props;

    return (
      <div className="travel-advisory-list">
        {travelAdvisories && (
          <div>
            {travelAdvisories.map((advisory, index) => (
              <TravelAdvisoryItem
                key={index}
                title={advisory.advisoryTitle}
                onClick={this._showTravelAdvisoryDetails.bind(this, index)}
              />
            ))}
          </div>
        )}
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  travelAdvisories: state.app.travelAdvisory.messageTravelAdvisory
});

const mapDispatchToProps = {
  getTravelAdvisoriesFn: TravelAdvisoryActions.getTravelAdvisories
};

const enhancers = _.flowRight(withConnectedReactRouter, connect(mapStateToProps, mapDispatchToProps));

export default enhancers(TravelAdvisoryListPage);
