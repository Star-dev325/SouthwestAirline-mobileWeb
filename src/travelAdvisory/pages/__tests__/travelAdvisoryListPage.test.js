import React from 'react';
import { render } from '@testing-library/react';

import { TravelAdvisoryListPage } from 'src/travelAdvisory/pages/travelAdvisoryListPage';

describe('TravelAdvisoryListPage', () => {
  let getTravelAdvisoriesFnStub;
  let pushStub;

  beforeEach(() => {
    getTravelAdvisoriesFnStub = jest.fn().mockResolvedValue();
    pushStub = jest.fn();
  });

  it('should render 3 items with title when there are 3 advisories', () => {
    const { container } = createComponent({
      travelAdvisories: [
        {
          id: 'idnumber1',
          advisoryTitle: 'title1',
          advisoryInfo: 'info1'
        },
        {
          id: 'idnumber2',
          advisoryTitle: 'title2',
          advisoryInfo: 'info2'
        },
        {
          id: 'idnumber3',
          advisoryTitle: 'title3',
          advisoryInfo: 'info3'
        }
      ]
    });

    expect(container).toMatchSnapshot();
  });

  it('should render no items when there are no advisories', () => {
    const { container } = createComponent({ travelAdvisories: [] });
    const items = container.querySelectorAll('.travel-advisory-item--title');

    expect(items.length).toEqual(0);
  });

  const createComponent = (props = {}) => {
    const defaultProps = {
      getTravelAdvisoriesFn: getTravelAdvisoriesFnStub,
      push: pushStub
    };

    const travelAdvisoryProps = { ...defaultProps, ...props };

    return render(<TravelAdvisoryListPage { ...travelAdvisoryProps}  />);
  };
});
