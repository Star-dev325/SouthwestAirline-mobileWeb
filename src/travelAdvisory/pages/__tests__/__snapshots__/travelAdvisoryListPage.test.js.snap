// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravelAdvisoryListPage should render 3 items with title when there are 3 advisories 1`] = `
<div>
  <div
    class="travel-advisory-list"
  >
    <div>
      <div
        class="nav-item travel-advisory-item"
      >
        <i
          class="travel-advisory-item--title-icon icon icon_exclamation-circle"
        />
        <span
          class="travel-advisory-item--title"
        >
          title1
        </span>
        <i
          class="icon icon_keyboard-arrow-right"
        />
      </div>
      <div
        class="nav-item travel-advisory-item"
      >
        <i
          class="travel-advisory-item--title-icon icon icon_exclamation-circle"
        />
        <span
          class="travel-advisory-item--title"
        >
          title2
        </span>
        <i
          class="icon icon_keyboard-arrow-right"
        />
      </div>
      <div
        class="nav-item travel-advisory-item"
      >
        <i
          class="travel-advisory-item--title-icon icon icon_exclamation-circle"
        />
        <span
          class="travel-advisory-item--title"
        >
          title3
        </span>
        <i
          class="icon icon_keyboard-arrow-right"
        />
      </div>
    </div>
  </div>
</div>
`;
